registry/storage/driver/s3-aws/s3_test.go

```diff
@@ -500,6 +500,7 @@ func TestWalk(t *testing.T) {
 	}
 
 	// create file structure matching fileset above
+	// nolint:prealloc
```

Comment:
Why are we not pre allocating this one?

Reply:
Technically we could allocate the capacity of `len(fileset)`, yeah, but in case `drvr.PutContent` (https://github.com/distribution/distribution/blob/59fd8656ac3c4f34de9dbb93e9fae90f7f1ec9f2/registry/storage/driver/s3-aws/s3_test.go#L506) fails and returns `error` we'd have some extra memory allocated, though maybe in this case (test with known size of the slice being iterated), that's fine.

Reply:
Addressed: https://github.com/distribution/distribution/pull/4037/commits/108980064305fc67affd9cb45886e828aac9563a